File: sample/kenviro/controller/DecisionSupportSystemController.js
var DecisionSupportSystemController = function DecisionSupportSystemController(cfg)
{
    let ctrl = {};

    ctrl.Submit = function (data, onResponse)
    {
        cfg.Connector.Submit(data, "ardvro/component/kenviro/DecisionSupportSystem/Submit", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Import = function (aknSubjectId, name, decisionMethodType, list, pfeProfileId, onResponse)
    {
        let args = {
            aisCriteriaId: aknSubjectId,
            pfeProfileId: pfeProfileId,
            Name: name,
            DecisionType: decisionMethodType,
            List: list
        };
        cfg.Connector.Submit(args, "ardvro/component/kenviro/DecisionSupportSystem/Import", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GetById = function (id, callback)
    {
        cfg.Connector.Submit(id, "ardvro/component/kenviro/DecisionSupportSystem/GetById", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Delete = function (data, callback)
    {
        cfg.Connector.Submit(data, "ardvro/component/kenviro/DecisionSupportSystem/Delete", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
};

File: sdk/.net/ardvro.sdk.net/component/kenviro/controller/DecisionSupportSystemController.cs
namespace ardvro.sdk.net.component.kenviro.controller;

using ardvro.adapter.ais.ent;
using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class DecisionSupportSystemController
{
    private IConnection _connection;
    public DecisionSupportSystemController(IConnection connection)
    {
        _connection = connection;
    }

    public void Submit(aisDss data, Action<aisDss> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/DecisionSupportSystem/Submit", callback);
    }

    public void Import(int aisCriteriaId, String name, String decisionMethodType, List<Flexible> list, int pfeProfileId, Action<aisDss> callback)
    {
        Flexible data = new Flexible();
        data.Add("aisCriteriaId", aisCriteriaId);
        data.Add("pfeProfileId", pfeProfileId);
        data.Add("Name", name);
        data.Add("DecisionType", decisionMethodType);
        data.Add("List", list);

        _connection.Submit(data, "ardvro/component/kenviro/DecisionSupportSystem/Import", callback);
    }

    public void GetById(int id, Action<aisDss> callback)
    {
        _connection.Submit(new Object[] { id }, "ardvro/component/kenviro/DecisionSupportSystem/GetById", callback);
    }

    public void Delete(aisDss data, Action<aisDss> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/DecisionSupportSystem/Delete", callback);
    }

}
